src/java/voldemort/server/Host.java

```diff
@@ -0,0 +1,26 @@
+package voldemort.server;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+import voldemort.VoldemortApplicationException;
+
+
+public class Host {
+
+    private static InetAddress getLocal() {
+        try {
+            return InetAddress.getLocalHost();
+        } catch(UnknownHostException ex) {
+            throw new VoldemortApplicationException("Error retrieving local host", ex);
+        }
+    }
+
+    public static String getName() {
+        return getLocal().getHostName();
+    }
+
+    public static String getFQDN() {
+        return getLocal().getCanonicalHostName();
+    }
```

Comment:
Executing hte code on my linux desktop one gives getHostName give machinename. getCanonicalHostName gives machinename.domain.extention

On our production clusters, both return the same though.